Quotient/room.cpp

```diff
@@ -623,7 +623,8 @@ QStringList Room::pinnedEventIds() const {
 QVector<const Quotient::RoomEvent*> Quotient::Room::pinnedEvents() const
 {
     QVector<const RoomEvent*> pinnedEvents;
-    for (const auto& evtId : pinnedEventIds())
+    const auto eventIds = pinnedEventIds();
+    for (const auto& evtId : eventIds)
```

Comment:
```suggestion
    for (const auto& evtId : std::as_const(eventIds))
```

Reply:
this doesn't work `error: use of deleted function ‘void std::as_const(const _Tp&&) [with _Tp = QStringList]’`

std::as_const only works with non const local variable, not with temporary

Reply:
Hm, indeed.